packages/contracts/contracts/libs/v0.8.x/minter-libs/RAMLib.sol

```diff
@@ -0,0 +1,2923 @@
+            // CHECKS
+            // require bidder is non-zero address (i.e. bid exists)
+            address bidderAddress = bid.bidder;
+            require(bidderAddress != address(0), "invalid Bid ID");
+            // if bid is already minted or refunded, skip to next bid
+            // @dev do not revert, since this could be due to front-running
+            if (
+                _getBidPackedBool(bid, INDEX_IS_MINTED) ||
+                _getBidPackedBool(bid, INDEX_IS_REFUNDED)
+            ) {
+                continue;
+            }
+            // require sender is bidder if requireSenderIsBidder is true
+            if (requireSenderIsBidder) {
+                require(msg.sender == bidderAddress, "Only sender is bidder");
+            }
+            // EFFECTS
+            // Settle and Refund the Bid
+            _settleAndRefundBid({
+                projectId: projectId,
+                coreContract: coreContract,
+                projectPrice: projectPrice,
+                slotIndex: bid.slotIndex,
+                bidId: currentBidId,
+                minterRefundGasLimit: minterRefundGasLimit
+            });
+            numRefundsIssued++;
+        }
+        // update number of bids refunded
+        RAMProjectConfig_.numBidsErrorRefunded += numRefundsIssued;
+    }
+
+    /**
+     * @notice Function to automatically refund the lowest winning bids for
+     * project `projectId` on core contract `coreContract` to resolve error
+     * state E1.
+     * Reverts if project is not in post-auction state C.
+     * Reverts if project is not in error state E1.
+     * Reverts if numBidsToRefund exceeds the number of bids that need to be
+     * refunded to resolve the error state E1.
+     * @dev Recommend admin-only not for security, but rather to enable Admin
+     * to be aware that an error state has been encountered while in post-
+     * auction state C.
+     * @param projectId Project ID to refunds bids for.
+     * @param coreContract Core contract address for the given project.
+     * @param numBidsToRefund Number of bids to refund in this call.
+     * @param minterRefundGasLimit Gas limit to use when refunding the bidder.
+     */
+    function autoRefundBidsToResolveE1(
+        uint256 projectId,
+        address coreContract,
+        uint24 numBidsToRefund,
+        uint256 minterRefundGasLimit
+    ) internal {
+        // CHECKS
+        // @dev block scope to limit stack depth
+        {
+            // require project minter state C (Post-Auction, admin-artist-only,
+            //  not all bids handled)
+            ProjectMinterStates projectMinterState = getProjectMinterState({
+                projectId: projectId,
+                coreContract: coreContract
+            });
+            require(
+                projectMinterState ==
+                    ProjectMinterStates.PostAuctionSellOutAdminArtistMint,
+                "Only post-auction admin-artist-only"
```

Comment:
nit: make <32 chars for contract size efficiency

Reply:
🙏